test/browser.multiapp.test.js

```diff
@@ -79,28 +83,37 @@ function browserifyTestApps(apps, next) {
         appRootDir: appDir,
       };
     }
-    boot.compileToBrowserify(opts, b);
+    bundles.push(opts);
   }
-
-  exportBrowserifyToFile(b, 'browser-app-bundle.js', next);
+  async.eachSeries(bundles, function(opts, done) {
+    boot.compileToBrowserify(opts, b, done);
+  }, function(err) {
+    exportBrowserifyToFile(b, 'browser-app-bundle.js', next);
+  });
 }
 
-function executeBundledApps(bundlePath, apps) {
+function executeBundledApps(bundlePath, apps, done) {
   var code = fs.readFileSync(bundlePath);
   var context = createBrowserLikeContext();
   vm.runInContext(code, context, bundlePath);
 
+  var ids = [];
   var script = 'var apps = {};\n';
   for (var i in apps) {
     var moduleName = apps[i].moduleName;
     var id = apps[i].appId || 'defaultApp';
+    ids.push(id);
     script += 'apps.' + id + ' = require("' + moduleName + '");\n';
   }
   script += 'apps;\n';
 
   var appsInContext = vm.runInContext(script, context);
-
-  printContextLogs(context);
+  async.each(ids, function(id, done) {
+    appsInContext[id].start(done);
+  }, function(err) {
+    printContextLogs(context);
+    done();
+  });
 
   return appsInContext;
```

Comment:
An async function that also returns a value in sync way is very weird. Please pass `appsInContext` to the callback.

```js
async.each(ids, function(id, done) {
  appsInContext[id].start(done);
}, function(err) {  
  printContextLogs(context);
  done(err, appsInContext);
});
```